File: android/api-tests/src/test/java/com/revenuecat/apitests/java/OnResultAnyApiTests.java
package com.revenuecat.apitests.java;

import com.revenuecat.purchases.hybridcommon.ErrorContainer;
import com.revenuecat.purchases.hybridcommon.OnResultAny;

@SuppressWarnings("unused")
class OnResultAnyApiTests<T> {
    private void checkOnSuccess(OnResultAny<T> onResult,
                                T result) {
        onResult.onReceived(result);
    }

    private void checkOnError(OnResultAny<T> onResult,
                              ErrorContainer errorContainer) {
        onResult.onError(errorContainer);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/OnResultAnyApiTests.kt
package com.revenuecat.apitests.kotlin

import com.revenuecat.purchases.hybridcommon.ErrorContainer
import com.revenuecat.purchases.hybridcommon.OnResultAny

@Suppress("unused")
private class OnResultAnyApiTests<T> {
    fun checkOnSuccess(
        onResult: OnResultAny<T>,
        result: T,
    ) {
        onResult.onReceived(result)
    }

    fun checkOnError(
        onResult: OnResultAny<T>,
        errorContainer: ErrorContainer,
    ) {
        onResult.onError(errorContainer)
    }
}
